elasticdl/proto/elasticdl.proto

```diff
@@ -53,6 +53,9 @@ message Tensor {
     // Indices will be tf.IndexedSlices.indices if the tensor is in the form
     // of tf.IndexedSlices. Ohterwise indices will be None.
     repeated int32 indices = 3;
+
+    // Dtype of the tensor, e.g. "int64", "float32".
+    string dtype = 4;
```

Comment:
Data type should be an enum type, string is too flexible.

Reply:
`enum` is clear. `string` can support arbitrary types (supported by Numpy) and simple to code.

Reply:
Use enum type instead.